### PR TITLE
Stencil buffer support and some more

### PR DESCRIPTION
**Pull request by digetx**
https://github.com/grate-driver/envytools/pull/5

### CODE REVIEW

File: include/tgr_3d.xml.h [hunk 1]
```diff
@@ -4,11 +4,11 @@
 /* Autogenerated file, DO NOT EDIT manually!
 
 This file was generated by the rules-ng-ng headergen tool in this git repository:
-http://github.com/envytools/envytools/
-git clone https://github.com/envytools/envytools.git
+http://0x04.net/cgit/index.cgi/rules-ng-ng
+git clone git://0x04.net/rules-ng-ng
```
Comment (kusma) [inline include/tgr_3d.xml.h]: Huh, what happened here?

File: include/tgr_3d.xml.h [hunk 2]
```diff
@@ -91,6 +91,22 @@ WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
...
+#define STENCIL_OP_INCR_WRAP					0x00000006
+#define STENCIL_OP_DECR_WRAP					0x00000007
+#define STENCIL_FUNC_NEVER					0x00000000
+#define STENCIL_FUNC_LESS					0x00000001
+#define STENCIL_FUNC_EQUAL					0x00000002
+#define STENCIL_FUNC_LESS_EQUAL					0x00000003
+#define STENCIL_FUNC_GREATER					0x00000004
+#define STENCIL_FUNC_NOTEQUAL					0x00000005
+#define STENCIL_FUNC_GREATER_EQUAL				0x00000006
+#define STENCIL_FUNC_ALWAYS					0x00000007
```
Comment (kusma) [inline include/tgr_3d.xml.h]: Hmm, looks a bit like we should namespace this, seem kinda risky for collisions (and inconsistent with other things here)

File: include/tgr_3d.xml.h (hunk 1 above)
Comment (digetx) [reply include/tgr_3d.xml.h]: Seems envytools doesn't re-gen the header entirely on each update. I removed old tgr_3d.xml.h and it is fine now.

File: include/tgr_3d.xml.h (hunk 2 above)
Comment (digetx) [reply include/tgr_3d.xml.h]: What about to prepend  them with TGR3D_ ?

File: include/tgr_3d.xml.h (hunk 1 above)
Comment (kusma) [reply include/tgr_3d.xml.h]: But these new urls are wrong, aren't they?

File: include/tgr_3d.xml.h (hunk 2 above)
Comment (kusma) [reply include/tgr_3d.xml.h]: Yeah, that sounds reasonable to me...

File: include/tgr_3d.xml.h (hunk 1 above)
Comment (digetx) [reply include/tgr_3d.xml.h]: They are old :) And seem to be wrong now.

**Review by kusma — APPROVED** [review]
Looks good!

**Pull request merged**